### PR TITLE
Fix missing pugixml dependency in generated CMake package file

### PR DESCRIPTION
**Pull request by myd7349**
libxdf depends on `pugixml`, but this dependency is missing in the generated CMake package file. As a result, users must have this in their CMakeLists.txt:

```cmake
find_package(libxdf CONFIG REQUIRED)
find_package(pugixml CONFIG REQUIRED) # This line is necessary as `pugixml` dependency is missing in the generated CMake package files
add_executable(demo PRIVATE XDF::xdf)
```

With this patch, the users of libxdf do not need to handle `pugixml` dependency by themselves:

```cmake
find_package(libxdf CONFIG REQUIRED)
add_executable(demo PRIVATE XDF::xdf)
```

And this pattern has been adopted by many widely used libraries like:

- https://github.com/nanomsg/nng/blob/master/cmake/nng-config.cmake.in
- https://github.com/facebook/folly/blob/main/CMake/folly-config.cmake.in

### CODE REVIEW

File: libxdfConfig.cmake.in [hunk 1]
```diff
@@ -0,0 +1,9 @@
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+
+if(NOT TARGET XDF::xdf)
+    include("${CMAKE_CURRENT_LIST_DIR}/libxdfTargets.cmake")
+endif()
+
+find_dependency(pugixml)
```
Comment (dg0yt) [inline libxdfConfig.cmake.in:9]: This line should be moved up. If `pugixml` isn't found, `include("${CMAKE_CURRENT_LIST_DIR}/libxdfTargets.cmake")` shall not be called.

Comment (myd7349) [reply libxdfConfig.cmake.in]: Hi! @dg0yt Thanks for your review. That sounds reasonable to me.

**Comment by Yida-Lin** [comment]
Thanks! Some of the workflow did not go through; could you please help take a look.

**Comment by Yida-Lin** [comment]
Sounds good, thanks @myd7349 and @dg0yt !

**Pull request merged**